src/lib/core/usecase/case-study-usecase.ts

```diff
@@ -0,0 +1,395 @@
+
+    if (caseStudySourceDataWithoutMap.length === 0) {
+      this.presenter.presentError({
+        status: "error",
+        operation: "usecase#case-study",
+        message: "No source data found for the case study.",
+        context: {
+          caseStudyName,
+          tracerID,
+          jobID,
+        },
+      });
+      return;
+    }
+
+    const agentSourceData = caseStudySourceDataWithoutMap.concat(explicitAgentSourceData);
+
+    const agentRemoteFiles: RemoteFile[] = agentSourceData.filter((file): file is RemoteFile => file.type === "remote");
+
+    if (agentRemoteFiles.length === 0) {
+      this.presenter.presentError({
+        status: "error",
+        operation: "usecase#case-study",
+        message: "After filering, no source data was left to create the agent.",
+        context: {
+          caseStudyName,
+          tracerID,
+          jobID,
+        },
+      });
+      return;
+    }
+
+    this.presenter.presentProgress({
+      status: "progress",
+      message: "Source data prepared. Creating vector store...",
+      context: {
+        caseStudyName,
+        tracerID,
+        jobID,
+      },
+    });
+
+    // 6. Create a new VectorStore with (C), then create a new Agent as usual
+    const createVectorStoreDTO = await this.vectorStore.createVectorStore(agentRemoteFiles);
+
+    if (!createVectorStoreDTO.success) {
+      this.presenter.presentError({
+        status: "error",
+        operation: "usecase#case-study",
+        message: "Failed to create vector store for the provided files. Check again later.",
+        context: createVectorStoreDTO,
+      });
+      return;
+    }
+
+    const vectorStoreID = createVectorStoreDTO.data.id;
+    const nonVectorStoreFiles = createVectorStoreDTO.data.unsupportedFiles;
+
+    this.presenter.presentProgress({
+      status: "progress",
+      message: "Vector store created. Creating agent...",
+      context: createVectorStoreDTO,
+    });
+
+    const researchContextDescripttion = `This Research Context was created for the case study ${caseStudyName}, with tracer ID ${tracerID} and job ID ${jobID}. It contains source data that complements the map that the users see next to the conversation with the agent.`;
+
+    const agentSystemInstructions = `You are a research assistant for the case study with name '${caseStudyName}'. You have been provided, through a vector store, with data that complements the map that users see next to the conversation with you. This data is meant to help you understand the context of the conversation better, and to provide more accurate and relevant responses to the queries. The source of this data is different scrapers that have parsed different APIs relevant to the case study in question. So, when asked about any file that you should have access to, please consider that it may be in the vector store.`;
```

Comment:
@maany this will override the default agent system instructions (which in this branch are now exposed to be changed). I guess this should be good enough, but we can modify it if needed